src/main/java/mate/academy/springboot/swagger/service/ProductServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.springboot.swagger.service;
+
+import java.math.BigDecimal;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.springboot.swagger.model.Product;
+import mate.academy.springboot.swagger.repository.ProductRepository;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ProductServiceImpl implements ProductService {
+    private final ProductRepository productRepository;
+
+    @Override
+    public Product create(Product product) {
+        return productRepository.save(product);
+    }
+
+    @Override
+    public Product findById(Long id) {
+        return productRepository.findById(id).orElseThrow(
+                () -> new RuntimeException("Can't find product with id: " + id)
+        );
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        productRepository.deleteById(id);
+    }
+
+    @Override
+    public Product update(Product product) {
+        return productRepository.save(product);
+    }
+
+    @Override
+    public List<Product> getAll(PageRequest pageRequest) {
+        return productRepository.findAll(pageRequest).toList();
+    }
+
+    @Override
+    public List<Product> getAllByPriceBetween(BigDecimal from,
+                                              BigDecimal to, PageRequest pageRequest) {
```

Comment:
There two common styles people use usually:
```suggestion
    public List<Product> getAllByPriceBetween(BigDecimal from,
                                              BigDecimal to, 
                                              PageRequest pageRequest) {
```
and 
```suggestion
    public List<Product> getAllByPriceBetween(
        BigDecimal from,
        BigDecimal to, 
        PageRequest pageRequest
    ) {
```